grammar.js

```diff
@@ -949,6 +951,17 @@ module.exports = grammar(C, {
       repeat1(choice($.raw_string_literal, $.string_literal))
     ),
 
+    user_defined_literal: $ => prec(PREC.USER_DEFINED_LITERAL, seq(
+      field('literal', choice(
+        $.number_literal,
+        $.char_literal,
+        $.string_literal,
+        $.raw_string_literal,
+        $.concatenated_string
+      )),
+      field('suffix', token.immediate(/[a-zA-Z_]\w*/))
```

Comment:
```suggestion
      field('suffix', token.immediate($.identifier))
```

Reply:
I tried that, but it causes the following error:

```
Error processing rule user_defined_literal_token1

Caused by:
    Grammar error: Unexpected rule Symbol(Symbol { kind: NonTerminal, index: 131 })
```

It seems `token.immediate()` interferes with both this and the `suffix` named field you mentioned in the other comment.

Reply:
Could you create another rule (name it whatever you wish, this is an example):

```js
udl_suffix: $ => token.immediate(/[a-zA-Z_]\w*/),
```

And use that as the suffix field in this rule (instead of a named `field` since it would be redundant)? It should at least make the token a named node in the syntax tree.

Reply:
Didn't realise I could do that. Done!